flocker/control/test/test_httpapi.py

```diff
@@ -2968,6 +2968,44 @@ def test_one_container(self):
             b"GET", b"/state/containers", None, OK, response
         )
 
+    def test_one_container_maximum_size(self):
+        """
+        When the cluster state includes one container that has a dataset with
+        a maximum size, the endpoint returns a single-element list
+        containing the container.
+        """
+        manifestation = Manifestation(
+            dataset=Dataset(dataset_id=unicode(uuid4()),
+                            maximum_size=1234),
+            primary=False
+        )
+        expected_application = Application(
+            name=u"myapp", image=DockerImage.from_string(u"busybox:1.2"),
+            volume=AttachedVolume(manifestation=manifestation,
+                                  mountpoint=FilePath(b"/xxx/yyy")),
+        )
+        expected_hostname = u"192.0.2.101"
+        self.cluster_state_service.update_node_state(
+            NodeState(
+                hostname=expected_hostname,
+                applications={expected_application},
+                manifestations={manifestation},
```

Comment:
Presumably meant to be `{manifestation.dataset_id: manifestation}` seeing as `manifestations` is a `PMap`